theia-tree-editor/src/browser/tree-editor-widget.tsx

```diff
@@ -80,6 +81,18 @@ export abstract class BaseTreeEditorWidget extends BaseWidget implements Saveabl
         this.configureTitle(this.title);
     }
 
+    get saveable(): Saveable {
+        return this;
+    }
+
+    createSnapshot(): object {
```

Comment:
Is there a reasonable default implementation we could provide here or in the `ResourceTreeEditorWidget`? (Same for `revert`)?